lib/std/compress/zstandard.zig

```diff
@@ -0,0 +1,275 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const RingBuffer = std.RingBuffer;
+
+const types = @import("zstandard/types.zig");
+pub const frame = types.frame;
+pub const compressed_block = types.compressed_block;
+
+pub const decompress = @import("zstandard/decompress.zig");
+
+pub fn Stream(
```

Comment:
It might be better to name this `DecompressStream` or `Decompressor` to future-proof it if there's ever a compressor implementation added.

Note, though, that this naming is currently inconsistent across the different decompressors in the std lib now, so there's no real convention to follow yet.
- `deflate` names it `Decompressor`
- Some just name it `Decompress` (and the helper function is `decompress` which conflicts with the current `decompress` function in `std.compress.zstandard`)
- `zlib` names it `ZlibStream` and https://github.com/ziglang/zig/pull/13977 was going to name it `ZlibStreamReader` [or something like it](https://github.com/ziglang/zig/pull/13977#discussion_r1096595299))

Reply:
I'll switch it to `DecompressStream` - I don't like just `Decompress` as it give no indication that it's a streaming interface.

Reply:
Though maybe things could just be named `std.compress.[algorithm].Reader` and `std.compress.[algorithm].Writer` if they are just providing a reader/writer interface.